virttest/qemu_vm.py

```diff
@@ -1100,6 +1100,7 @@ def add_memorys(devices, params):
                 if params.get("hugepage_path"):
                     backend_options["backend_mem"] = "memory-backend-file"
                     backend_options["mem-path_mem"] = params["hugepage_path"]
+                    backend_options["prealloc_mem"] = params.get("prealloc_mem", "yes")
```

Comment:
`prealloc` is a common option for memory objects, not only for hugepage.
```
  prealloc-context=<link<thread-context>> - Context to use for creating CPU threads for preallocation
  prealloc-threads=<int> - Number of CPU threads to use for prealloc
  prealloc=<bool>        - Preallocate memory
```

You can set default to yes for hugepage, but also need to consider the `vm_mem`.

Reply:
@PaulYuuu sorry I don't get your point, what's exactly your suggestion? Is it related with the [comment below](https://github.com/avocado-framework/avocado-vt/pull/3765#discussion_r1320928240)?

Reply:
Yes @mcasquer, `memory-backend-ram`, `memory-backend-file` and `memory-backend-memfd` are support `prealloc`, so you can use `vm_mem_prealloc` to set it. But for `hugepage`, I think directly set `backend_options["prealloc_mem"] = "yes"` is okay as suggested  by the developer. Or you can also make if as configurable: `backend_options["prealloc_mem"] = params.get("vm_mem_prealloc", "yes")
`

Reply:
> Yes @mcasquer, `memory-backend-ram`, `memory-backend-file` and `memory-backend-memfd` are support `prealloc`, so you can use `vm_mem_prealloc` to set it. But for `hugepage`, I think directly set `backend_options["prealloc_mem"] = "yes"` is okay as suggested by the developer. Or you can also make if as configurable: `backend_options["prealloc_mem"] = params.get("vm_mem_prealloc", "yes") `

@PaulYuuu got it, thanks for the explanation :grinning: